### ISSUE TITLE
Grid does not render with Angular 6 in production mode

### ISSUE THREAD

**Issue by mjurisic** [ag-grid/ag-grid-angular#212]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid-angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

ag-grid-angular does not render and throws one or more of the following errors in javascript console after upgrading to angular 6.0.0:
Cannot read property 'COLUMN' of undefined
Cannot read property 'ROW' of undefined
Cannot read property 'AG-GRID' of undefined 

happens only in production build, dev build seems to be working ok

**Expected behavior**

grid with data should show on page without errors in the console
**Minimal reproduction of the problem with instructions**

I created a fresh Angular CLI project with dummy ag-grid to reproduce the problem:  
https://github.com/mjurisic/angular6-ag-grid

Steps to reproduce:
git clone https://github.com/mjurisic/angular6-ag-grid.git
cd angular6-ag-grid
npm install
ng build --prod
cd dist
http-server

**What is the motivation / use case for changing the behavior?**

We upgraded our enterprise application to the last Angular, ag-grid etc. and all grids stopped working.

**Please tell us about your environment:**

Windows 7, Webstorm/VisualStudio Code, npm, node 10
* **ag-grid-angular version:** 17.1.0

* **ag-Grid/ag-Grid-Enterprise version:** 17.1.1

* **Angular version:** angular version: 6.0.0

* **Browser:** [all]

* **Language:** [TypeScript 2.7.2]

**Comment by core-avatar** [comment]
Deactivating buildOptimizer aka TreeShake "temporarily fixes" the issue at cost of 5x the package size...

**Comment by Alex-Torres** [comment]
Having same exact issue.

**Comment by mikeberezowski** [comment]
I'm experiencing this too. Followed the Angular-CLI tutorial and ran ng-serve --prod and still get the same error.

**Comment by weihe8892** [comment]
Upgrade to angular/cli 6.0.1 seems fix the issue.

**Comment by mjurisic** [comment]
Upgrading to Angular 6.0.1 with "@angular-devkit/build-angular": "0.6.1" and "@angular-devkit/build-optimizer": "0.6.1" has solved the issue.

**Comment by werthdavid** [comment]
Not resolved for me with Angular 6.0.3 😢
